pkg/k8shandler/prometheus_rule.go

```diff
@@ -47,17 +46,15 @@ func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdatePrometheusRules(
 
 	current := &monitoringv1.PrometheusRule{}
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		err = elasticsearchRequest.client.Get(ctx, types.NamespacedName{Name: rule.Name, Namespace: rule.Namespace}, current)
+		err = er.client.Get(ctx, types.NamespacedName{Name: rule.Name, Namespace: rule.Namespace}, current)
 		if err != nil {
-			logrus.Debugf("could not get prometheus rule %q: %v", rule.Name, err)
 			return err
 		}
 
 		current.Spec = rule.Spec
-		if err = elasticsearchRequest.client.Update(ctx, current); err != nil {
+		if err = er.client.Update(ctx, current); err != nil {
 			return err
 		}
-		logrus.Debug("updated prometheus rules")
```

Comment:
if we did a better job about only performing updates when we need i would say this would be useful to have at Info level... 
(we don't do a comparison to see if the specs are different to even warrant doing an `update`)